Bluepill-cli/Bluepill-cli/Simulator/SimulatorMonitor.m

```diff
@@ -135,6 +137,10 @@ - (void)onTestCaseFailedWithName:(NSString *)testName inClass:(NSString *)testCl
      [currentTime timeIntervalSinceDate:self.lastTestCaseStartDate],
      fullTestName];
 
+    // Save screenshot for failed test
+    NSString *screenshotName = [fullTestName stringByReplacingOccurrencesOfString:@"/" withString:@"_"];
+    [self.screenshotService saveScreenshotForFailedTestWithName:screenshotName];
```

Comment:
It is inside BPTreeParser, so whenever an assertion failed, `regex = [NSRegularExpression regularExpressionWithPattern:TEST_CASE_FAILED options:0 error:nil];` will be called but we don't call the `onTestCaseFailedWithName` method till the test case is failed in the end.

Yah, it is known that CI is yet stable for our tests :(